### PR TITLE
test: Collect code coverage in the "devel" scenario

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] https://github.com/cockpit-project/cockpit/pull/17078
- [ ] bots needs to be updated to include the 'devel' scenario in the auto-triggered tests.

### CODE REVIEW

**Comment by martinpitt** [comment]
Moving to draft, as this is conflicty. Unfortunately the test logs are gone, so hard to say what was wrong.

**Comment by jelly** [comment]
The double box shadow should be undone with a custom class which unsets it.

**Comment by jelly** [comment]
Unconflicted let's get this landed :)

**Comment by jelly** [comment]
devel passes, so let's merge this and then merge a bots PR https://github.com/cockpit-project/cockpit-machines/pull/650

**Review by KKoukiou — APPROVED** [review]

**Review by KKoukiou — COMMENTED** [review]
@jelly when I click on the coverage link from https://cockpit-logs.us-east-1.linodeobjects.com/pull-650-20230608-184147-4798ec01-fedora-38-devel/log.html there is not a valid link though.

**Comment by jelly** [comment]
Somehow coverage is not generated, this needs debugging.

**Comment by jelly** [comment]
Coverage seems to fail to generate due to

```
SKIP http://127.0.0.2:9891/cockpit/$49506e66ea0aea98a2fd97e00f6b6807eea7c6c7ea9b35c1af7f55453ffaff47/machines/index.js -> /work/bots/make-checkout-workdir/dist/index.js
```

So it seems the `dist_map` is bogus

**Comment by jelly** [comment]
```
distmap {'machines': '/work/bots/make-checkout-workdir/dist'}
```

This looks good, yet we still skip:

```
SKIP http://127.0.0.2:9891/cockpit/$49506e66ea0aea98a2fd97e00f6b6807eea7c6c7ea9b35c1af7f55453ffaff47/machines/index.js -> /work/bots/make-checkout-workdir/dist/index.js
```

So most likely is that the `.map` file is missing

**Comment by jelly** [comment]
So.. we basically don't use `NODE_ENV=devel`.

File: src/helpers.js [hunk 1]
```diff
@@ -871,5 +871,6 @@ export function getIfaceSourceName(iface) {
 }
 
 export function canDeleteDiskFile(disk) {
+    console.log(disk);
```
Comment (cockpituous) [inline src/helpers.js:874]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-650-20230609-135838-a45ed3fa-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

**Comment by jelly** [comment]
Yay it works now https://cockpit-logs.us-east-1.linodeobjects.com/pull-650-20230609-140812-3430f06e-fedora-38-devel/Coverage/index.html

**Comment by jelly** [comment]
Rawhide is also borked in https://github.com/cockpit-project/cockpit-machines/pull/1104

**Pull request merged**